libdnf/dnf-repo.cpp

```diff
@@ -1907,6 +1907,7 @@ dnf_repo_update(DnfRepo *repo,
     dnf_state_release_locks(state);
     lr_handle_setopt(priv->repo_handle, NULL, LRO_PROGRESSCB, NULL);
     lr_handle_setopt(priv->repo_handle, NULL, LRO_PROGRESSDATA, 0xdeadbeef);
+    lr_handle_setopt(priv->repo_handle, NULL, LRO_HMFCB, NULL);
```

Comment:
LRO_HMFCB uses the same user data as LRO_PROGRESSCB.
Please reset LRO_HMFCB before userdata (LRO_PROGRESSDATA) are modified. ->
move your line before `lr_handle_setopt(priv->repo_handle, NULL, LRO_PROGRESSDATA, 0xdeadbeef);`